### PR TITLE
Fix complex parameter initialization

### PR DESCRIPTION
**Pull request by Shaikh-Ubaid**
Also update the error message.

fixes https://github.com/lfortran/lfortran/issues/3098

### CODE REVIEW

**Comment by Shaikh-Ubaid** [comment]
Updated error message:
```bash
(lf2) % cat examples/expr2.f90 
program expr2
implicit none

    integer :: x = 10
    complex(8), parameter :: zero = (x, x)
    print *, zero

end program
(lf2) % gfortran examples/expr2.f90 
examples/expr2.f90:5:39:

    5 |     complex(8), parameter :: zero = (x, x)
      |                                       1
Error: Expected a right parenthesis in expression at (1)
examples/expr2.f90:6:17:

    6 |     print *, zero
      |                 1
Error: Symbol 'zero' at (1) has no IMPLICIT type
(lf2) % lfortran examples/expr2.f90
semantic error: Initialisation of `zero` must reduce to a compile time constant.
 --> examples/expr2.f90:5:5
  |
5 |     complex(8), parameter :: zero = (x, x)
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ 


Note: Please report unclear or confusing messages as bugs at
https://github.com/lfortran/lfortran/issues.
```

File: src/lfortran/semantics/ast_common_visitor.h [hunk 1]
```diff
@@ -1944,8 +1944,8 @@ class CommonVisitor : public AST::BaseVisitor<Derived> {
                                     }
                                     ASR::expr_t* init_expr_value = ASRUtils::expr_value(init_expr);
                                     if( init_expr && !ASRUtils::is_value_constant(init_expr_value) ) {
-                                        throw SemanticError("Initialisation of " + std::string(x.m_syms[i].m_name) +
-                                                            " must reduce to a compile time constant.",
+                                        throw SemanticError("Initialisation of `" + std::string(x.m_syms[i].m_name) +
```
Comment (certik) [inline src/lfortran/semantics/ast_common_visitor.h:1947]:
```suggestion
                                        throw SemanticError("Initialization of `" + std::string(x.m_syms[i].m_name) +
```

File: src/lfortran/semantics/ast_common_visitor.h [hunk 2]
```diff
@@ -2403,8 +2403,8 @@ class CommonVisitor : public AST::BaseVisitor<Derived> {
                         } else if( ASRUtils::is_value_constant(init_expr) ) {
                             value = nullptr;
                         } else {
-                            throw SemanticError("Initialisation of " + std::string(x.m_syms[i].m_name) +
-                                                " must reduce to a compile time constant.",
+                            throw SemanticError("Initialisation of `" + std::string(x.m_syms[i].m_name) +
```
Comment (certik) [inline src/lfortran/semantics/ast_common_visitor.h:2406]:
```suggestion
                            throw SemanticError("Initialization of `" + std::string(x.m_syms[i].m_name) +
```

File: src/lfortran/semantics/ast_common_visitor.h [hunk 3]
```diff
@@ -2455,8 +2455,8 @@ class CommonVisitor : public AST::BaseVisitor<Derived> {
                         */
                         value = ASRUtils::expr_value(init_expr);
                         if (value == nullptr) {
-                            throw SemanticError("Initialisation of " + std::string(x.m_syms[i].m_name) +
-                                                " must reduce to a compile time constant.",
+                            throw SemanticError("Initialisation of `" + std::string(x.m_syms[i].m_name) +
```
Comment (certik) [inline src/lfortran/semantics/ast_common_visitor.h:2458]:
```suggestion
                            throw SemanticError("Initialization of `" + std::string(x.m_syms[i].m_name) +
```

File: src/lfortran/semantics/ast_common_visitor.h [hunk 4]
```diff
@@ -2470,8 +2470,8 @@ class CommonVisitor : public AST::BaseVisitor<Derived> {
                                 type = ASRUtils::TYPE(ASR::make_Character_t(al, int_const->base.base.loc, 1, len, nullptr));
                             }
                         } else {
-                            throw SemanticError("Initialisation of " + std::string(x.m_syms[i].m_name) +
-                                                " must reduce to a compile time constant.",
+                            throw SemanticError("Initialisation of `" + std::string(x.m_syms[i].m_name) +
```
Comment (certik) [inline src/lfortran/semantics/ast_common_visitor.h:2473]:
```suggestion
                            throw SemanticError("Initialization of `" + std::string(x.m_syms[i].m_name) +
```

File: src/lfortran/semantics/ast_common_visitor.h [hunk 5]
```diff
@@ -2521,8 +2521,8 @@ class CommonVisitor : public AST::BaseVisitor<Derived> {
                                     value = nullptr;
                                 }
                             } else {
-                                throw SemanticError("Initialisation of " + std::string(x.m_syms[i].m_name) +
-                                                    " must reduce to a compile time constant.",
+                                throw SemanticError("Initialisation of `" + std::string(x.m_syms[i].m_name) +
```
Comment (certik) [inline src/lfortran/semantics/ast_common_visitor.h:2524]:
```suggestion
                                throw SemanticError("Initialization of `" + std::string(x.m_syms[i].m_name) +
```

**Review by certik — APPROVED** [review]
I think it looks great, thanks! I left some minor spelling correction.

**Pull request merged**